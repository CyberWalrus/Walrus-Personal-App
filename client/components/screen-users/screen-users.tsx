import RoutePath, { getRoute } from "@client/routes";
import { User } from "@client/type/data";
import { PureComponent, ReactElement } from "react";
import * as React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { Operation } from "../../store/data/data";
import { getUsers } from "../../store/data/selectors";
import { StateApp, ThunkDispatch } from "../../type/reducer";

interface PropsState {
  users: User[];
}
interface PropsDispatch {
  onUsersLoad: () => void;
}
type Props = PropsState & PropsDispatch;

class ScreenUsers extends PureComponent<Props> {
  public componentDidMount(): void {
    this.props.onUsersLoad();
  }
  public render(): ReactElement {
    const { users }: Props = this.props;
    return (
      <main className={`page-content`}>
        <section className={`form-custom-box`}>
          {users &&
            users.map((item: User, index: number) => (
              <Link to={getRoute(item.id, RoutePath.USER_LOGIN)} key={index}>
                <div>{item.id}</div>
                <div>{item.login}</div>
                <div>{item.email}</div>
              </Link>
            ))}
        </section>
      </main>
    );
  }
}

const mapStateToProps = (state: StateApp, ownProps: Props): Props => ({
  ...ownProps,
  users: getUsers(state),
});

const mapDispatchToProps = (dispatch: ThunkDispatch): PropsDispatch => ({
  onUsersLoad: (): void => {
    dispatch(Operation.getUsers());
  },
});

export { ScreenUsers };

export default connect<Props, PropsDispatch, {}, StateApp>(
  mapStateToProps,
  mapDispatchToProps,
)(ScreenUsers);
