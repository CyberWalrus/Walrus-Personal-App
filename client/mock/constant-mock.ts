import { FormType } from "../../config/constants";
import { Option } from "../hocs/with-form-state/with-form-state";
import { InputCustomInterface } from "../type/component";
import { User, UserRole } from "../type/data";

// tslint:disable-next-line:no-empty
const _functVoid = (): void => {};

const FORM_TYPE: FormType = FormType.SIGN_IN;

const user: User = {
  id: `test`,
  email: `test`,
  isActive: true,
  nickName: `test`,
  login: `test`,
  firstName: `test`,
  lastName: `test`,
  userRoleId: [`test`],
  signUpDate: 1234567,
  created: 1234567,
};
const userRole: UserRole = {
  id: `test`,
  name: `test`,
  isActive: true,
};
const optionForm: Option = {
  values: {},
  formErrors: {},
  formValid: true,
  onChangeUserInput: _functVoid,
  onClickSubmit: _functVoid,
};
const optionInput: InputCustomInterface = {
  formTypes: [FormType.SIGN_UP],
  type: `text`,
  name: `login`,
  placeHolder: `Login`,
  hidenValue: `Login`,
};

const usersEmpty: User[] = [];
const users: User[] = [user, user];

const userRolesEmpty: UserRole[] = [];
const userRoles: UserRole[] = [userRole, userRole];

export {
  user,
  usersEmpty,
  users,
  optionForm,
  FORM_TYPE,
  optionInput,
  userRole,
  userRolesEmpty,
  userRoles,
};
