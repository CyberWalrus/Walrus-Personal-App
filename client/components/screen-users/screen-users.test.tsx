import * as Enzyme from "enzyme";
import { shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson, { Json } from "enzyme-to-json";
import * as React from "react";
import { users } from "../../mock/constant-mock";
import { ScreenUsers } from "./screen-users";

Enzyme.configure({ adapter: new Adapter() });
it(`ScreenUsers correctly render without data`, (): void => {
  const jestFunc = jest.fn();
  const tree: Json = toJson(
    shallow(<ScreenUsers users={[]} onUsersLoad={jestFunc} />),
  );
  expect(tree).toMatchSnapshot();
});
it(`ScreenUsers correctly render with data`, (): void => {
  const jestFunc = jest.fn();
  const tree: Json = toJson(
    shallow(<ScreenUsers users={users} onUsersLoad={jestFunc} />),
  );
  expect(tree).toMatchSnapshot();
});
