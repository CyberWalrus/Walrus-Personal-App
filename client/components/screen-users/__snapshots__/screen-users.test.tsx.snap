// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenUsers correctly render with data 1`] = `
<main
  className="page-content"
>
  <section
    className="form-custom-box"
  >
    <Link
      key="0"
      to="/user/test"
    >
      <div>
        test
      </div>
      <div>
        test
      </div>
      <div>
        test
      </div>
    </Link>
    <Link
      key="1"
      to="/user/test"
    >
      <div>
        test
      </div>
      <div>
        test
      </div>
      <div>
        test
      </div>
    </Link>
  </section>
</main>
`;

exports[`ScreenUsers correctly render without data 1`] = `
<main
  className="page-content"
>
  <section
    className="form-custom-box"
  />
</main>
`;
